cmd/nodelink-controller/main.go

```diff
@@ -393,8 +393,22 @@ func (c *Controller) processNode(node *corev1.Node) error {
 		modNode.Labels[k] = v
 	}
 
-	// Taints are to be an authoritative list on the machine spec per cluster-api comments:
```

Comment:
Nice!
Can you add a PR upstream even if it's to update the field comment and why we think should be additive?
Can you please add a validation for this machine/node taint reconciliation here https://github.com/openshift/machine-api-operator/blob/master/test/e2e/operator_expectations.go?

Reply:
sure. how about handling e2e in  follow-up PR?

Reply:
please let's include it in this one, the point of having it in the same PR is that it prevent us from merging code which does not work as expected rather than having to fix later